src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationServiceImpl.java

```diff
@@ -58,8 +61,11 @@ public AuthenticationServiceImpl(AuthenticationTokenRepository authenticationTok
     public String createNewAuthToken(String email) {
         User user = userService.getUserByEmail(email);
 
-        // Delete the old Token
-        authenticationTokenRepository.findByUserEmail(email).ifPresent(authenticationTokenRepository::delete);
+        // Delete the old Tokens if there are more than 4.
+        List<AuthenticationToken> existingTokens = authenticationTokenRepository.findByUserEmail(email);
+        if (existingTokens.size() >= 4) {
+            authenticationTokenRepository.deleteAll(existingTokens);
```

Comment:
So if there are more than 4 tokens, we're deleting them all? Makes more sense to only delete the oldest right? Deleting them all introduces a sort of random factor for the users. One time they are still logged in, and the other time all devices are signed out.

Reply:
That's a solid point! I will change it.